src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al

```diff
@@ -0,0 +1,187 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.TestLibraries.Apps.AppSource;
+
+using System.Environment.Configuration;
+using System.RestClient;
+using System.Apps.AppSource;
+
+/// <summary>
+/// Library for managing AppSource product retrival and usage.
+/// </summary>
+codeunit 132910 "AppSource Product Manager Test" implements "AppSource Product Manager Dependencies"
```

Comment:
This code would be much easier to understand if you split it into 2 codeunits: 
- one is the implementation of the interface, which is really only used for mocking some env. stuff in the tests
- the other is the 'test library' which would be helper methods to ease testing and some wrappers around internal functionality